<?xml version='1.0' encoding='UTF-8'?>
<root>
 <tabbed_widget name="Main Window" parent="main_window">
  <Tab containers="1" tab_name="tab1">
   <Container>
    <DockSplitter count="2" sizes="0.500561;0.499439" orientation="-">
     <DockArea name="...">
      <plot style="Lines" flip_x="false" mode="TimeSeries" flip_y="false">
       <range bottom="-0.025000" left="5.927300" right="25.926700" top="1.025000"/>
       <limitY/>
       <curve name="/base_est_rt/base_estimation_node/contact_state" color="#1f77b4"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot style="Lines" flip_x="false" mode="TimeSeries" flip_y="false">
       <range bottom="-2.504258" left="5.927300" right="25.926700" top="102.674563"/>
       <limitY/>
       <curve name="/base_est_rt/base_estimation_node/vertex_weights.0" color="#d62728"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="tab2">
   <Container>
    <DockSplitter count="3" sizes="0.332584;0.334831;0.332584" orientation="-">
     <DockSplitter count="2" sizes="0.5;0.5" orientation="|">
      <DockArea name="...">
       <plot style="Lines" flip_x="false" mode="TimeSeries" flip_y="false">
        <range bottom="-0.000000" left="5.927300" right="25.926700" top="0.000000"/>
        <limitY/>
        <curve name="/base_est_rt/base_estimation_node/wrench.0" color="#1ac938"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" flip_x="false" mode="TimeSeries" flip_y="false">
        <range bottom="-0.100000" left="5.927300" right="25.926700" top="0.100000"/>
        <limitY/>
        <curve name="/base_est_rt/base_estimation_node/wrench.3" color="#9467bd"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter count="2" sizes="0.5;0.5" orientation="|">
      <DockArea name="...">
       <plot style="Lines" flip_x="false" mode="TimeSeries" flip_y="false">
        <range bottom="-1268.006819" left="5.927300" right="25.926700" top="471.697603"/>
        <limitY/>
        <curve name="/base_est_rt/base_estimation_node/wrench.1" color="#ff7f0e"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" flip_x="false" mode="TimeSeries" flip_y="false">
        <range bottom="-0.100000" left="5.927300" right="25.926700" top="0.100000"/>
        <limitY/>
        <curve name="/base_est_rt/base_estimation_node/wrench.4" color="#17becf"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter count="2" sizes="0.5;0.5" orientation="|">
      <DockArea name="...">
       <plot style="Lines" flip_x="false" mode="TimeSeries" flip_y="false">
        <range bottom="-5060.739180" left="5.927300" right="25.926700" top="11449.385617"/>
        <limitY/>
        <curve name="/base_est_rt/base_estimation_node/wrench.2" color="#f14cc1"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" flip_x="false" mode="TimeSeries" flip_y="false">
        <range bottom="-0.100000" left="5.927300" right="25.926700" top="0.100000"/>
        <limitY/>
        <curve name="/base_est_rt/base_estimation_node/wrench.5" color="#bcbd22"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="tab2">
   <Container>
    <DockSplitter count="1" sizes="1" orientation="-">
     <DockArea name="...">
      <plot style="Lines" flip_x="false" mode="TimeSeries" flip_y="false">
       <range bottom="-4.089428" left="5.927300" right="25.926700" top="21.096165"/>
       <limitY/>
       <curve name="/base_est_rt/base_estimation_node/base_vel_meas.2" color="#1f77b4"/>
       <curve name="/base_est_rt/base_estimation_node/passive_jnt_v_est" color="#d62728"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="tab3">
   <Container>
    <DockSplitter count="1" sizes="1" orientation="-">
     <DockArea name="...">
      <plot style="Lines" flip_x="false" mode="TimeSeries" flip_y="false">
       <range bottom="-23043.936844" left="5.927300" right="25.926700" top="24368.135437"/>
       <limitY/>
       <curve name="/base_est_rt/base_estimation_node/tau_c_raw.0" color="#1ac938"/>
       <curve name="/base_est_rt/base_estimation_node/tau_c_raw_filt.0" color="#ff7f0e"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <currentTabIndex index="3"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="1"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <default time_axis="" delimiter="0"/>
  </plugin>
  <plugin ID="DataLoad MCAP"/>
  <plugin ID="DataLoad ROS bags">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS Topic Subscriber">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="UDP Server"/>
  <plugin ID="WebSocket Server"/>
  <plugin ID="ZMQ Subscriber"/>
  <plugin ID="Fast Fourier Transform"/>
  <plugin ID="Quaternion to RPY"/>
  <plugin ID="Reactive Script Editor">
   <library code="--[[ Helper function to create a series from arrays&#xa;&#xa; new_series: a series previously created with ScatterXY.new(name)&#xa; prefix:     prefix of the timeseries, before the index of the array&#xa; suffix_X:   suffix to complete the name of the series containing the X value. If [nil], use the index of the array.&#xa; suffix_Y:   suffix to complete the name of the series containing the Y value&#xa; timestamp:   usually the tracker_time variable&#xa;              &#xa; Example:&#xa; &#xa; Assuming we have multiple series in the form:&#xa; &#xa;   /trajectory/node.{X}/position/x&#xa;   /trajectory/node.{X}/position/y&#xa;   &#xa; where {N} is the index of the array (integer). We can create a reactive series from the array with:&#xa; &#xa;   new_series = ScatterXY.new(&quot;my_trajectory&quot;) &#xa;   CreateSeriesFromArray( new_series, &quot;/trajectory/node&quot;, &quot;position/x&quot;, &quot;position/y&quot;, tracker_time );&#xa;--]]&#xa;&#xa;function CreateSeriesFromArray( new_series, prefix, suffix_X, suffix_Y, timestamp )&#xa;  &#xa;  --- clear previous values&#xa;  new_series:clear()&#xa;  &#xa;  --- Append points to new_series&#xa;  index = 0&#xa;  while(true) do&#xa;&#xa;    x = index;&#xa;    -- if not nil, get the X coordinate from a series&#xa;    if suffix_X ~= nil then &#xa;      series_x = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_X) )&#xa;      if series_x == nil then break end&#xa;      x = series_x:atTime(timestamp)&#x9; &#xa;    end&#xa;    &#xa;    series_y = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_Y) )&#xa;    if series_y == nil then break end &#xa;    y = series_y:atTime(timestamp)&#xa;    &#xa;    new_series:push_back(x,y)&#xa;    index = index+1&#xa;  end&#xa;end&#xa;&#xa;--[[ Similar to the built-in function GetSeriesNames(), but select only the names with a give prefix. --]]&#xa;&#xa;function GetSeriesNamesByPrefix(prefix)&#xa;  -- GetSeriesNames(9 is a built-in function&#xa;  all_names = GetSeriesNames()&#xa;  filtered_names = {}&#xa;  for i, name in ipairs(all_names)  do&#xa;    -- check the prefix&#xa;    if name:find(prefix, 1, #prefix) then&#xa;      table.insert(filtered_names, name);&#xa;    end&#xa;  end&#xa;  return filtered_names&#xa;end&#xa;&#xa;--[[ Modify an existing series, applying offsets to all their X and Y values&#xa;&#xa; series: an existing timeseries, obtained with TimeseriesView.find(name)&#xa; delta_x: offset to apply to each x value&#xa; delta_y: offset to apply to each y value &#xa;  &#xa;--]]&#xa;&#xa;function ApplyOffsetInPlace(series, delta_x, delta_y)&#xa;  -- use C++ indeces, not Lua indeces&#xa;  for index=0, series:size()-1 do&#xa;    x,y = series:at(index)&#xa;    series:set(index, x + delta_x, y + delta_y)&#xa;  end&#xa;end&#xa;"/>
   <scripts/>
  </plugin>
  <plugin ID="CSV Exporter"/>
  <plugin ID="ROS /rosout Visualization"/>
  <plugin ID="ROS Topic Re-Publisher"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets/>
 <!-- - - - - - - - - - - - - - - -->
</root>

